components/org.wso2.is.data.sync.client/src/main/java/org/wso2/is/data/sync/system/pipeline/DataSyncPipeline.java

```diff
@@ -110,12 +110,11 @@ public boolean processBatch() throws SyncClientException {
                         targetConnection.rollback();
                         sourceConnection.rollback();
                     } catch (SQLException e) {
-                        batchComplete = false;
                         log.error("Error while rolling back sync transaction on table: " + pipelineConfiguration
                                 .getTableName(), e);
                     }
                 }
-                complete = batchComplete && transactionResults.isEmpty();
+                complete = transactionResults.isEmpty() || !batchProcessingSuccess;
```

Comment:
remove loop here. change complete logic to check transactionResults sixe < batch size.